### PR TITLE
Use project structure favored by `pod lib create`

### PR DESCRIPTION
**Pull request by irace**
Reorganize the project to use the structure suggested by `pod lib create` (https://github.com/tumblr/XExtensionItem/issues/10).

Did my best to keep the git history sane here, though didn't worry as much about the files in the example project looking like `git mv`.

All tests should pass, sample project (both application and extension target) should run as expected.

### CODE REVIEW

File: Example/Tests/Tests-Info.plist [hunk 1]
```diff
@@ -7,13 +7,11 @@
 	<key>CFBundleDevelopmentRegion</key>
 	<string>en</string>
 	<key>CFBundleExecutable</key>
-	<string>$(EXECUTABLE_NAME)</string>
+	<string>${EXECUTABLE_NAME}</string>
 	<key>CFBundleIdentifier</key>
-	<string>com.tumblr.$(PRODUCT_NAME:rfc1034identifier)</string>
+	<string>org.cocoapods.demo.${PRODUCT_NAME:rfc1034identifier}</string>
```
Comment (mattbischoff) [inline Example/Tests/Tests-Info.plist]: This should probably go back.

File: Example/XExtensionItem/XExtensionItem-Info.plist [hunk 2]
```diff
@@ -4,14 +4,16 @@
...
 	<key>CFBundleDevelopmentRegion</key>
 	<string>en</string>
+	<key>CFBundleDisplayName</key>
+	<string>${PRODUCT_NAME}</string>
 	<key>CFBundleExecutable</key>
-	<string>$(EXECUTABLE_NAME)</string>
+	<string>${EXECUTABLE_NAME}</string>
 	<key>CFBundleIdentifier</key>
-	<string>com.tumblr.$(PRODUCT_NAME:rfc1034identifier)</string>
+	<string>org.cocoapods.demo.${PRODUCT_NAME:rfc1034identifier}</string>
```
Comment (mattbischoff) [inline Example/XExtensionItem/XExtensionItem-Info.plist]: Same.

File: Example/Tests/Tests-Info.plist (hunk 1 above)
Comment (irace) [inline Example/Tests/Tests-Info.plist]: Hm. This is what `pod lib create` gives out of the box. Not sure if it matters but guessing people don't generally change it?

Comment (orta) [inline Example/Tests/Tests-Info.plist]: Yeah, I'd presume that too, can't hurt either way

**Comment by irace** [comment]
This should be good to go now :dizzy:

**Pull request merged**